cmd/daemon/main.go

```diff
@@ -1,8 +1,63 @@
 package main
 
-import "fmt"
+import (
+	"context"
+	"log"
+	"net/http"
+
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"knative.dev/container-freezer/pkg/daemon"
+	"knative.dev/container-freezer/pkg/freeze/containerd"
+	pkglogging "knative.dev/pkg/logging"
+)
+
+const runtimeTypeContainerd string = "containerd"
 
 func main() {
-	// TODO delete -- temporarily added to merge hack project
-	fmt.Println("hello, freezer!")
+	logger, _ := pkglogging.NewLogger("", "")
+	runtimeType := runtimeTypeContainerd // TODO read from envvar
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var freezeThaw daemon.FreezeThawer
+	switch runtimeType {
+	case runtimeTypeContainerd:
+		logger.Info("creating new containerd freezeThawer")
+		freezeThaw, err = containerd.New()
+		// TODO suport docker, crio
+	default:
+		log.Fatal("unrecognised runtimeType", runtimeType)
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	http.ListenAndServe(":9696", &daemon.Handler{
```

Comment:
I think this needs to be :8080 to match containerPort in the yaml (the :9696 happens via the hostPort mapping to the containerPort)